jest-assessment-week-2-borja/code-challenges.test.js

```diff
@@ -0,0 +1,315 @@
+
+// pseudo: 
+// I will declare a function capitalized that accepts a parameter of an array of words
+// Function will use .map() function to iterate through the array and change each string within to uppercase the first letter
+// Use conditional statements if/else it to return divisible ot not divisible
+// input: array of strings
+// output: array of same length with same values with their first character capitalized
+// yarn jest should Pass (green)
+
+// I dind't know about the .substring method, but I found it in our lessons and looked up how it works at https://www.w3schools.com/jsref/jsref_substring.asp
+
+const capitalized = (array) => {
+    return array.map((value) => {
+        return value[0].toUpperCase() + value.substring(1) 
+    })
+}
+
+// PASS  Desktop/week-2-assessment-borja-x13/jest-assessment-week-2-borja/code-challenges.test.js
+// Tests:       2 passed, 2 total
+
+
+
+// --------------------3) Create a function that takes in a string and logs the index of the first vowel.
+
+// a) Create a test with expect statements for each of the variables provided.
+
+const vowelTester1 = "learn"
+// Expected output: 1
+const vowelTester2 = "academy"
+// Expected output: 0
+const vowelTester3 = "challenges"
+// Expected output: 2
+
+
+// pseudo code:
+// I will use the jest test template to test a function "vowelIndex"
+// it "takes a string and returns the first letter of each word"
+// expect(vowelIndex("learn")).toEqual(1)
+// expect(vowelIndex("academy")).toEqual(0)
+// expect(vowelIndex("challenges")).toEqual(2)
+// input: since there isn't a function to test, there is no input and I expect a fail (red)
+// output: should output a ReferenceError: vowelIndex is not defined
+
+describe ("vowelIndex", () => {
+    it ("Takes a string and logs the index of the first vowel in the string", () => {
+        expect(vowelIndex(vowelTester1)).toEqual(1)
+        expect(vowelIndex(vowelTester2)).toEqual(0)
+        expect(vowelIndex(vowelTester3)).toEqual(2)
+    })
+})
+
+// FAIL  Desktop/week-2-assessment-borja-x13/jest-assessment-week-2-borja/code-challenges.test.js
+// ● vowelIndex › Takes a string and logs the index of the first vowel in the string
+//   ReferenceError: vowelIndex is not defined
+
+
+// b) Create the function that makes the test pass.
+
+// pseudo: 
+// I will declare a function vowelIndex that accepts a parameter of string
+// use for ... in  loop to iterate through the string
+// Use conditional statement to compare the letter at the given index with a, e, i, o, u 
+// return the index of the first vowel using .indexOf() method
+// input: string
+// output: index of the first vowel found in the given string
+// yarn jest should Pass (green)
+
+const vowelIndex = (string) => {
+    for (let i in string) {
+        if(string[i] === "a" || string[i] === "e" || string[i] === "i" || string[i] === "o" || string[i] === "u") {
+           return string.indexOf(string[i])
+        }
+    }
+}
```

Comment:
Awesome job! Now we can start to consider what we want to happen if there is no vowel in the string.